prerendering.bs

```diff
@@ -309,6 +309,35 @@ Every {{Document}} has a <dfn for="Document">post-prerendering activation steps
 Every {{Document}} has an <dfn for="Document">activation start time</dfn>, which is initially a {{DOMHighResTimeStamp}} with a time value of zero.
 <p class="note">This is used to keep the time immediately after [=Prompt to unload|prompting to unload=] the [=active document=] of the previous [=browsing context=].
 
+
+<div algorithm="User-agent initiated prerendering">
+  User-agents may choose to initiate prerendering without a predecessor document, for example as a result of the address bar or other browser user interactions.
+
+  To <dfn export>Start user-agent initiated prerendering</dfn> given a [=URL=] |url|:
+
+  1. [=Assert=]: |startingURL|'s [=url/scheme=] is an [=HTTP(S) scheme=].
+
+  1. Let |bc| be the result of [=creating a new top-level browsing context=].
+
+  1. Set |bc|'s [=browsing context/loading mode=] to "`prerender`".
+
+  1. Let |request| be a new [=request=] whose [=request/URL=] is |startingURL| and [=request/referrer policy=] is "`no-referrer`".
+
+  1. [=Navigate=] |bc| to |request|.
+
+  1. Let |activate| be to do the following [=in parallel=]:
+
+    1. Let |unsafe time| be the [=unsafe shared current time=].
+
+    1. Update the user agent's user interface to present |bc|, e.g., by updating the tab/window contents and the browser chrome.
+
+    1. [=prerendering browsing context/Finalize activation=] for |bc| given |url|'s [=url/origin=] and |unsafe time|.
+
+  1. Return |activate|.
+
+    <p class="note">The returned algorithm steps can be used by the user-agent when the user is ready to "commit" the navigation.
```

Comment:
Since there are no actual callers, I think it's a bit clearer to change this so that there's no return value, but instead the step that currently reads "Let activate be to do the following in parallel:" to instead be "When the user indicates they wish to commit the navigation:"

We could also add some notes like "NOTE: an example of indicating commitment is pressing the Enter key in the URL bar", and "NOTE: the user might never indicate such a commitment, or might take long enough to do so that the user agent needs to reclaim the resources used by the prerender for some more immediate task. In that case the user agent can TODO" where the TODO stands in for our future definition of discarding prerendering BCs.

Reply:
Amended